### PR TITLE
Setup ngrok for mWeb

### PR DESCRIPTION
**Pull request by roryabraham**
### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/155133

### Tests
1. First, follow the instructions in [this SO](https://stackoverflow.com/c/expensify/questions/3382) and set up a new reserved subdomain, except instead of `expensify-name` use `expensify-cash-name`.
2. Then, run `cp ./config/ngrok.example.yml ./config/ngrok.yml`
3. Replace your name and auth token in `./config/ngrok.yml`
4. Update your `.env`:
    1) Change `NGROK_URL` to `NGROK_URL_COM`
    2) Change `USE_NGROK` to `USE_NGROK_COM`
    3) Add `NGROK_URL_CASH=https://expensify-cash-$REPLACE.ngrok.io`
    4) Add `USE_NGROK_CASH=true`
    5) If it's not already true, set `USE_NGROK_COM=true`
5. Run `npm run ngrok`. Make sure it's forwarding traffic from `expensify-$REPLACE.ngrok.io` -> `expensify.com.dev` and from `expensify-cash-$REPLACE.ngrok.io` -> `localhost:8080`. I had to run it twice for it to work.
6. In another terminal tab, run `npm run web`
7. On your mobile device, go to `https://expensify-cash-$REPLACE.ngrok.io`. It will take a little while on the first load, but then should be pretty fast (since the whole app bundle will be downloaded)
8. Sign in to the app (to make sure you can connect to the expensify.com.dev API).
9. Make a noticeable change to the code and save it. Wait a minute and see that that change appears via webpack's hot-reload. It's still a bit slow, but was faster than the initial load in my experience.
10. Kill the process running webpack (the terminal tab you ran `npm run web` in)
11. Change `USE_NGROK_CASH` to `false` in your `.env`
14. Re-run `npm run web` and visit `localhost:8080` in your browser on your laptop. Make sure the app loads.
15. Make a noticeable change to the code and save it. It should hot-reload the change in your browser window. You should still be seeing your API requests go through `ngrok`

### Tested On

- [x] Web
- [x] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

#### Web
image

### CODE REVIEW

**Comment by botify** [comment]
Hey, I noticed you changed some webpack configuration files. This can break production builds. Did you remember to run a production build locally to verify they still work?

**Comment by botify** [comment]
Hey, I noticed you changed some webpack configuration files. This can break production builds. Did you remember to run a production build locally to verify they still work?

**Comment by roryabraham** [comment]
I should probably also follow my own rule and run a local desktop build since I edited config files. As a follow-up I'll need to update [this SO](https://stackoverflow.com/c/expensify/questions/3382) and send out an `ACTION_REQUIRED` about the `.env` file changes.

**Comment by roryabraham** [comment]
smol bump

**Comment by AndrewGable** [comment]
Ok so this seems to add quite a bit of complexity to our somewhat simple ngrok setup. Could you help me understand something we'd need to test on a physical device that we can't test using Chrome debugger tools set to test on a mobile device? e.g. https://developer.chrome.com/docs/devtools/device-mode/

File: config/ngrok.example.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# To find your authtoken run `cat ~/.ngrok2/ngrok.yml | grep authtoken`
+authtoken: $REPLACE
```
Comment (thienlnam) [inline config/ngrok.example.yml:2]: NAB: I think a more familiar syntax would be <AUTH_TOKEN> and same with other replace instances in this file

**Review by thienlnam — COMMENTED** [review]
+1 to @AndrewGable comment above.  However, I did test this flow and it does work

**Comment by roryabraham** [comment]
You just can't test the physical UX of the mobile website using a simulator.

- One example might be haptic feedback, if we wanted to implement that.
- Another example came up when we were building a zoom-view for images on web – we needed to use ngrok to run the mobile website on a physical device test how that view interacts with the native "pinch-zoom" functionality offered by mobile devices.

Third, this is sort of a case of "you don't know what you don't know", but there are things you just might not think to test on a simulator that will just happen naturally on a physical device, such as swipe gestures and stuff.

I agree that this introduces some necessary degree of complexity, but I don't think it's outlandish to claim that it's _always_ better to test on a physical device rather than on a simulator, and this is meant to streamline that process and make it easier for everyone.

**Comment by roryabraham** [comment]
I'll concede that there's very little that _can't_ be tested on a simulator, but I'm still arguing that testing on physical devices is better. An alternate solution would be to just not merge this PR and let people set up a second dedicated ngrok URL only if they want to, but then webpack's hot reload won't work.

**Comment by AndrewGable** [comment]
I totally agree that testing physical is better, I just wonder how often that's actually done by developers. Personally I always use emulators until they don't support the feature I am trying to test: push notifications, apple pay, etc. Are there any other solutions that meet in the middle with less complexity, but still enable this testing?

**Comment by roryabraham** [comment]
Maybe there are ways to simplify it, but in summary this is what we have:

1) There are now two ngrok urls instead of one – no way to avoid this complexity AFAIK.
2) There's an ngrok config script which starts both tunnels at once. We could choose not to include this, but then people would have to run both ngrok tunnels (one manually, and one using the script in Expensidev). With the script & config created by this PR, it's just `npm run ngrok`

**Comment by roryabraham** [comment]
Seems like people weren't keen on this, so I'm going to close it out, along with the linked issue.

**Pull request closed without merging**